File: 2-Regression/1_Simple_Linear_Regression/src/simple_linear_regression.py
# Simple Linear Regression

# Import the Libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import os

from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression

# Import the data
Path = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'bin\Salary_Data.csv'))
dataset = pd.read_csv(Path)

X = dataset.iloc[:, :-1].values
Y = dataset.iloc[:, 1].values

# Splitting the dataset
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 1/3, random_state = 0)

# Fit Simple Linear Regression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)

# Predict the Test set
Y_pred = regressor.predict(X_test)

# Visualising the Training set
plt.scatter(X_train, Y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

# Visualising the Test set
plt.scatter(X_test, Y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Testing set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: 2-Regression/1-Simple_Linear_Regression/src/simple_linear_regression.r
# Simple Linear Regression

# Path to the file and Libraries
Path = 'MachineLearning-Portfolio/2-Regression/1-Simple_Linear_Regression/bin/Salary_Data.csv'
#install.packages('caTools', repos = "http://cran.us.r-project.org", dependencies = TRUE)
#install.packages("tidyverse", repos = "http://cran.us.r-project.org", dependencies = TRUE)
#install.packages("Hmisc", repos = "http://cran.us.r-project.org",  dependencies = TRUE)
#install.packages("lme4", repos = "http://cran.us.r-project.org",  dependencies = TRUE)
#install.packages('ggplot2', repos = "http://cran.us.r-project.org", dependencies = TRUE)
library(caTools, help, pos = 2, lib.loc = NULL)
library(ggplot2)
library(lme4)

# Import dataset
dataset = read.csv(Path)

# Splitting into Training and Test set
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Fitting Simple Linear Regression
regressor = lm(formula = Salary ~ YearsExperience, 
				data = training_set)

# Predicting the Test set data
Y_pred = predict(regressor, newdata = test_set)

# Visualization Training data
TrainingPlot <- ggplot() + 
	geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), 
		color = 'red') +
	geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
		color = 'blue') + 
	ggtitle('Salary vs Experience(Training set)') +
	xlab('Years of Experience') + 
	ylab('Salary')

print(TrainingPlot)

TestPlot <- ggplot() + 
	geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
		color = 'red') +
	geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), 
		color = 'blue') + 
	ggtitle('Salary vs Experience(Training set)') +
	xlab('Years of Experience') + 
	ylab('Salary')

print(TestPlot)